src/rendering/engine.rs

```diff
@@ -291,8 +295,7 @@ impl RenderingEngine {
                         translation + Translation::new(screen_size.0 / 2.0, screen_size.1 / 2.0);
                 }
 
-                translation.x += camera.offset.x;
-                translation.y += camera.offset.y;
+                translation += Translation::from(Vec2::from(camera.offset));
```

Comment:
I think instead of wrapping the offset into a `Vec2` we could just implement `From<nalgebra::Vector2<f32>>` for Translation